src/ccache.cpp

```diff
@@ -912,6 +912,11 @@ rewrite_stdout_from_compiler(const Context& ctx, std::string&& stdout_data)
   //
   // We don't want to cache those.
   if (!stdout_data.empty()) {
+    // Tokenizer is used with include_empty, so strip last \n to avoid a
```

Comment:
How about changing tokenizer mode to `skip_last_empty` instead?

Reply:
This could work, but there is still an issue when the output (or at least its last line) doesn't contain a newline at all. See the updated commit.

Reply:
Sure, handling that case as well would be good, even though it likely doesn't occur in practice.

Reply:
Done.